napari/_lazy.py

```diff
@@ -0,0 +1,67 @@
+import importlib
+import importlib.util
+
+
+def install_lazy(module_name, submodules=None, submod_attrs=None):
+    """Install lazily loaded submodules, and functions or other attributes.
+
+    Parameters
+    ----------
+    module_name : str
+        Typically use __name__.
+    submodules : set
+        List of submodules to install.
+    submod_attrs : dict
+        Dictionary of submodule -> list of attributes / functions.
+        These attributes are imported as they are used.
+
+    Returns
+    -------
+    __getattr__, __dir__, __all__
+
+    """
+    if submod_attrs is None:
+        submod_attrs = {}
+
+    if submodules is None:
+        submodules = set()
+    else:
+        submodules = set(submodules)
+
+    attr_to_modules = {
+        attr: mod for mod, attrs in submod_attrs.items() for attr in attrs
+    }
+
+    __all__ = list(submodules | attr_to_modules.keys())
+
+    def __getattr__(name):
+        # this unused import is here to fix a very strange bug.
+        # there is some mysterious magical goodness in scipy stats that needs
+        # to be imported early.
+        # see: https://github.com/napari/napari/issues/925
+        # see: https://github.com/napari/napari/issues/1347
+        from scipy import stats  # noqa: F401
```

Comment:
What is this about?

Reply:
Read the comments just above it. It's weird. 😂

Reply:
yeah, we honestly still don't know... the relationship with scipy.stats was first observed in https://github.com/napari/napari/issues/925#issuecomment-580059473... tried to take it out  in https://github.com/napari/napari/pull/1250, and then https://github.com/napari/napari/issues/1347 popped up again.  would be thrilled if you had any insighty